// @gohighlevel/api-client/dist/lib/code/campaigns/campaigns.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaigns = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Campaigns Service
 * Documentation for campaigns API
 */
class Campaigns {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Campaigns
     * Get Campaigns
     */
    async getCampaigns(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'status', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/campaigns/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Campaigns = Campaigns;
exports.default = Campaigns;
