// @gohighlevel/api-client/dist/lib/code/custom-fields/custom-fields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFields = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * CustomFields Service
 * Custom fields are data points that allow you to capture and store specific information tailored to your business requirements. You can create fields across field types like text, numeric, selection options and special fields like date/time or signature
 */
class CustomFields {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Custom Field / Folder By Id
     * &lt;div&gt;
                    &lt;p&gt; Get Custom Field / Folder By Id.&lt;/p&gt;
                    &lt;div&gt;
                      &lt;span style&#x3D; &quot;display: inline-block;
                                  width: 25px; height: 25px;
                                  background-color: yellow;
                                  color: black;
                                  font-weight: bold;
                                  font-size: 24px;
                                  text-align: center;
                                  line-height: 22px;
                                  border: 2px solid black;
                                  border-radius: 10%;
                                  margin-right: 10px;&quot;&gt;
                                  !
                        &lt;/span&gt;
                        &lt;span&gt;
                          &lt;strong&gt;
                          Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
                          &lt;/strong&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                  &lt;/div&gt;
     */
    async getCustomFieldById(params, options) {
        const paramDefs = [{ name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/custom-fields/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Custom Field By Id
     * &lt;div&gt;
      &lt;p&gt; Update Custom Field By Id &lt;/p&gt;
      &lt;div&gt;
        &lt;span style&#x3D; &quot;display: inline-block;
                    width: 25px; height: 25px;
                    background-color: yellow;
                    color: black;
                    font-weight: bold;
                    font-size: 24px;
                    text-align: center;
                    line-height: 22px;
                    border: 2px solid black;
                    border-radius: 10%;
                    margin-right: 10px;&quot;&gt;
                    !
          &lt;/span&gt;
          &lt;span&gt;
            &lt;strong&gt;
            Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
            &lt;/strong&gt;
          &lt;/span&gt;
      &lt;/div&gt;
    &lt;/div&gt;
     */
    async updateCustomField(params, requestBody, options) {
        const paramDefs = [{ name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/custom-fields/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Custom Field By Id
     * &lt;div&gt;
      &lt;p&gt; Delete Custom Field By Id &lt;/p&gt;
      &lt;div&gt;
        &lt;span style&#x3D; &quot;display: inline-block;
                    width: 25px; height: 25px;
                    background-color: yellow;
                    color: black;
                    font-weight: bold;
                    font-size: 24px;
                    text-align: center;
                    line-height: 22px;
                    border: 2px solid black;
                    border-radius: 10%;
                    margin-right: 10px;&quot;&gt;
                    !
          &lt;/span&gt;
          &lt;span&gt;
            &lt;strong&gt;
            Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
            &lt;/strong&gt;
          &lt;/span&gt;
      &lt;/div&gt;
    &lt;/div&gt;
     */
    async deleteCustomField(params, options) {
        const paramDefs = [{ name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/custom-fields/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Custom Fields By Object Key
     * &lt;div&gt;
                    &lt;p&gt; Get Custom Fields By Object Key&lt;/p&gt;
                    &lt;div&gt;
                      &lt;span style&#x3D; &quot;display: inline-block;
                                  width: 25px; height: 25px;
                                  background-color: yellow;
                                  color: black;
                                  font-weight: bold;
                                  font-size: 24px;
                                  text-align: center;
                                  line-height: 22px;
                                  border: 2px solid black;
                                  border-radius: 10%;
                                  margin-right: 10px;&quot;&gt;
                                  !
                        &lt;/span&gt;
                        &lt;span&gt;
                          &lt;strong&gt;
                          Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
                          &lt;/strong&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                  &lt;/div&gt;
     */
    async getCustomFieldsByObjectKey(params, options) {
        const paramDefs = [{ name: 'objectKey', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/custom-fields/object-key/{objectKey}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Custom Field Folder
     * &lt;div&gt;
      &lt;p&gt; Create Custom Field Folder &lt;/p&gt;
      &lt;div&gt;
        &lt;span style&#x3D; &quot;display: inline-block;
                    width: 25px; height: 25px;
                    background-color: yellow;
                    color: black;
                    font-weight: bold;
                    font-size: 24px;
                    text-align: center;
                    line-height: 22px;
                    border: 2px solid black;
                    border-radius: 10%;
                    margin-right: 10px;&quot;&gt;
                    !
          &lt;/span&gt;
          &lt;span&gt;
            &lt;strong&gt;
            Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
            &lt;/strong&gt;
          &lt;/span&gt;
      &lt;/div&gt;
    &lt;/div&gt;
     */
    async createCustomFieldFolder(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/custom-fields/folder', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Custom Field Folder Name
     * &lt;div&gt;
      &lt;p&gt; Create Custom Field Folder &lt;/p&gt;
      &lt;div&gt;
        &lt;span style&#x3D; &quot;display: inline-block;
                    width: 25px; height: 25px;
                    background-color: yellow;
                    color: black;
                    font-weight: bold;
                    font-size: 24px;
                    text-align: center;
                    line-height: 22px;
                    border: 2px solid black;
                    border-radius: 10%;
                    margin-right: 10px;&quot;&gt;
                    !
          &lt;/span&gt;
          &lt;span&gt;
            &lt;strong&gt;
            Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
            &lt;/strong&gt;
          &lt;/span&gt;
      &lt;/div&gt;
    &lt;/div&gt;
     */
    async updateCustomFieldFolder(params, requestBody, options) {
        const paramDefs = [{ name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/custom-fields/folder/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Custom Field Folder
     * &lt;div&gt;
      &lt;p&gt; Create Custom Field Folder &lt;/p&gt;
      &lt;div&gt;
        &lt;span style&#x3D; &quot;display: inline-block;
                    width: 25px; height: 25px;
                    background-color: yellow;
                    color: black;
                    font-weight: bold;
                    font-size: 24px;
                    text-align: center;
                    line-height: 22px;
                    border: 2px solid black;
                    border-radius: 10%;
                    margin-right: 10px;&quot;&gt;
                    !
          &lt;/span&gt;
          &lt;span&gt;
            &lt;strong&gt;
            Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
            &lt;/strong&gt;
          &lt;/span&gt;
      &lt;/div&gt;
    &lt;/div&gt;
     */
    async deleteCustomFieldFolder(params, options) {
        const paramDefs = [{ name: 'id', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/custom-fields/folder/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Custom Field
     * &lt;div&gt;
                    &lt;p&gt; Create Custom Field &lt;/p&gt;
                    &lt;div&gt;
                      &lt;span style&#x3D; &quot;display: inline-block;
                                  width: 25px; height: 25px;
                                  background-color: yellow;
                                  color: black;
                                  font-weight: bold;
                                  font-size: 24px;
                                  text-align: center;
                                  line-height: 22px;
                                  border: 2px solid black;
                                  border-radius: 10%;
                                  margin-right: 10px;&quot;&gt;
                                  !
                        &lt;/span&gt;
                        &lt;span&gt;
                          &lt;strong&gt;
                          Only supports Custom Objects and Company (Business) today. Will be extended to other Standard Objects in the future.
                          &lt;/strong&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                  &lt;/div&gt;
     */
    async createCustomField(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/custom-fields/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.CustomFields = CustomFields;
exports.default = CustomFields;
