// @gohighlevel/api-client/dist/lib/code/oauth/oauth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Oauth = void 0;
const constants_1 = require("../../constants");
const request_utils_1 = require("../../utils/request-utils");
/**
 * Oauth Service
 * Documentation for OAuth 2.0 API
 */
class Oauth {
    constructor(httpClient, config = {}) {
        this.MARKETPLACE_URL = 'https://marketplace.gohighlevel.com';
        this.client = httpClient;
        this.config = config;
    }
    /**
     * Generate OAuth authorization URL for the authorization code flow
     */
    getAuthorizationUrl(clientId, redirectUri, scope) {
        const params = {
            client_id: clientId,
            redirect_uri: redirectUri,
            scope,
            response_type: 'code'
        };
        return `${this.MARKETPLACE_URL}/oauth/chooselocation?${new URLSearchParams(params).toString()}`;
    }
    /**
     * Refresh access token using refresh token
     * @param refreshToken The refresh token
     * @param clientId OAuth client ID
     * @param clientSecret OAuth client secret
     * @param grantType Grant type (must be 'refresh_token')
     * @param userType User type (UserType.Location or UserType.Company)
     */
    async refreshToken(refreshToken, clientId, clientSecret, grantType, userType) {
        if (grantType !== 'refresh_token') {
            throw new Error('grantType must be "refresh_token"');
        }
        if (!Object.values(constants_1.UserType).includes(userType)) {
            throw new Error(`userType must be "${constants_1.UserType.Location}" or "${constants_1.UserType.Company}"`);
        }
        return this.getAccessToken({
            refresh_token: refreshToken,
            client_id: clientId,
            client_secret: clientSecret,
            grant_type: grantType,
            user_type: userType
        });
    }
    /**
     * Get Access Token
     * Use Access Tokens to access GoHighLevel resources on behalf of an authenticated location/company.
     */
    async getAccessToken(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = [];
        const processedBody = new URLSearchParams(requestBody).toString();
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/oauth/token', extracted.path),
            params: extracted.query,
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
                ...extracted.header,
                ...options?.headers
            },
            data: processedBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Location Access Token from Agency Token
     * This API allows you to generate locationAccessToken from AgencyAccessToken
     */
    async getLocationAccessToken(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Agency-Access-Only"];
        const processedBody = new URLSearchParams(requestBody).toString();
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/oauth/locationToken', extracted.path),
            params: extracted.query,
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
                ...extracted.header,
                ...options?.headers
            },
            data: processedBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Location where app is installed
     * This API allows you fetch location where app is installed upon
     */
    async getInstalledLocation(params, options) {
        const paramDefs = [{ name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'query', in: 'query' }, { name: 'isInstalled', in: 'query' }, { name: 'companyId', in: 'query' }, { name: 'appId', in: 'query' }, { name: 'versionId', in: 'query' }, { name: 'onTrial', in: 'query' }, { name: 'planId', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/oauth/installedLocations', extracted.path),
            params: extracted.query,
            headers: {
                ...extracted.header,
                ...options?.headers
            },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Oauth = Oauth;
exports.default = Oauth;
