// @gohighlevel/api-client/dist/lib/code/saas-api/saas-api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SaasApi = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * SaasApi Service
 * API Service for SaaS
 */
class SaasApi {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get locations by stripeId with companyId
     * Get locations by stripeCustomerId or stripeSubscriptionId with companyId
     * @deprecated This method is deprecated
     */
    async locationsDeprecated(params, options) {
        const paramDefs = [{ name: 'customerId', in: 'query' }, { name: 'subscriptionId', in: 'query' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/locations', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update SaaS subscription
     * Update SaaS subscription for given locationId and customerId
     * @deprecated This method is deprecated
     */
    async updateSaasSubscriptionDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/update-saas-subscription/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Disable SaaS for locations
     * Disable SaaS for locations for given locationIds
     * @deprecated This method is deprecated
     */
    async bulkDisableSaasDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/bulk-disable-saas/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Enable SaaS for Sub-Account (Formerly Location)
     * &lt;div&gt;
                    &lt;p&gt;Enable SaaS for Sub-Account (Formerly Location) based on the data provided&lt;/p&gt;
                    &lt;div&gt;
                      &lt;span style&#x3D; &quot;display: inline-block;
                                  width: 25px; height: 25px;
                                  background-color: yellow;
                                  color: black;
                                  font-weight: bold;
                                  font-size: 24px;
                                  text-align: center;
                                  line-height: 22px;
                                  border: 2px solid black;
                                  border-radius: 10%;
                                  margin-right: 10px;&quot;&gt;
                                  !
                        &lt;/span&gt;
                        &lt;span&gt;
                          &lt;strong&gt;
                            This feature is only available on Agency Pro ($497) plan.
                          &lt;/strong&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                  &lt;/div&gt;
      
     * @deprecated This method is deprecated
     */
    async enableSaasLocationDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/enable-saas/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Pause location
     * Pause Sub account for given locationId
     * @deprecated This method is deprecated Use Sub account for given locationId instead.
     */
    async pauseLocationDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/pause/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Rebilling
     * Bulk update rebilling for given locationIds
     * @deprecated This method is deprecated
     */
    async updateRebillingDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/update-rebilling/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Agency Plans
     * Fetch all agency subscription plans for a given company ID
     * @deprecated This method is deprecated
     */
    async getAgencyPlansDeprecated(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/agency-plans/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Location Subscription Details
     * Fetch subscription details for a specific location from location metadata
     * @deprecated This method is deprecated
     */
    async getLocationSubscriptionDeprecated(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/get-saas-subscription/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Bulk Enable SaaS
     * Enable SaaS mode for multiple locations with support for both SaaS v1 and v2
     * @deprecated This method is deprecated
     */
    async bulkEnableSaasDeprecated(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/bulk-enable-saas/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get SaaS Locations
     * Fetch all SaaS-activated locations for a company with pagination
     * @deprecated This method is deprecated
     */
    async getSaasLocationsDeprecated(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }, { name: 'page', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/saas-locations/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get SaaS Plan
     * Fetch a specific SaaS plan by plan ID
     * @deprecated This method is deprecated
     */
    async getSaasPlanDeprecated(params, options) {
        const paramDefs = [{ name: 'planId', in: 'path' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas-api/public-api/saas-plan/{planId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get locations by stripeId with companyId
     * Get locations by stripeCustomerId or stripeSubscriptionId with companyId
     */
    async locations(params, options) {
        const paramDefs = [{ name: 'customerId', in: 'query' }, { name: 'subscriptionId', in: 'query' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas/locations', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update SaaS subscription
     * Update SaaS subscription for given locationId and customerId
     */
    async generatePaymentLink(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/saas/update-saas-subscription/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Disable SaaS for locations
     * Disable SaaS for locations for given locationIds
     */
    async bulkDisableSaas(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas/bulk-disable-saas/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Enable SaaS for Sub-Account (Formerly Location)
     * &lt;div&gt;
                    &lt;p&gt;Enable SaaS for Sub-Account (Formerly Location) based on the data provided&lt;/p&gt;
                    &lt;div&gt;
                      &lt;span style&#x3D; &quot;display: inline-block;
                                  width: 25px; height: 25px;
                                  background-color: yellow;
                                  color: black;
                                  font-weight: bold;
                                  font-size: 24px;
                                  text-align: center;
                                  line-height: 22px;
                                  border: 2px solid black;
                                  border-radius: 10%;
                                  margin-right: 10px;&quot;&gt;
                                  !
                        &lt;/span&gt;
                        &lt;span&gt;
                          &lt;strong&gt;
                            This feature is only available on Agency Pro ($497) plan.
                          &lt;/strong&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                  &lt;/div&gt;
      
     */
    async enableSaasLocation(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas/enable-saas/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Pause location
     * Pause Sub account for given locationId
     */
    async pauseLocation(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas/pause/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Rebilling
     * Bulk update rebilling for given locationIds
     */
    async updateRebilling(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas/update-rebilling/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Agency Plans
     * Fetch all agency subscription plans for a given company ID
     */
    async getAgencyPlans(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas/agency-plans/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Location Subscription Details
     * Fetch subscription details for a specific location from location metadata
     */
    async getLocationSubscription(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas/get-saas-subscription/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Bulk Enable SaaS
     * Enable SaaS mode for multiple locations with support for both SaaS v1 and v2
     */
    async bulkEnableSaas(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/saas/bulk-enable-saas/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get SaaS Locations
     * Fetch all SaaS-activated locations for a company with pagination
     */
    async getSaasLocations(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }, { name: 'page', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas/saas-locations/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get SaaS Plan
     * Fetch a specific SaaS plan by plan ID
     */
    async getSaasPlan(params, options) {
        const paramDefs = [{ name: 'planId', in: 'path' }, { name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/saas/saas-plan/{planId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.SaasApi = SaasApi;
exports.default = SaasApi;
