// @gohighlevel/api-client/dist/lib/code/proposals/proposals.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Proposals = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Proposals Service
 * Documentation for Documents and Contracts API
 */
class Proposals {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * List documents
     * List documents for a location
     */
    async listDocumentsContracts(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'status', in: 'query' }, { name: 'paymentStatus', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'query', in: 'query' }, { name: 'dateFrom', in: 'query' }, { name: 'dateTo', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/proposals/document', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Send document
     * Send document to a client
     */
    async sendDocumentsContracts(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/proposals/document/send', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List templates
     * List document contract templates for a location
     */
    async listDocumentsContractsTemplates(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'dateFrom', in: 'query' }, { name: 'dateTo', in: 'query' }, { name: 'type', in: 'query' }, { name: 'name', in: 'query' }, { name: 'isPublicDocument', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'skip', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/proposals/templates', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Send template
     * Send template to a client
     */
    async sendDocumentsContractsTemplate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/proposals/templates/send', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Proposals = Proposals;
exports.default = Proposals;
