// @gohighlevel/api-client/dist/lib/code/associations/associations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Associations = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Associations Service
 * Documentation for Associations API
 */
class Associations {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Create Relation for you associated entities.
     * Create Relation.Documentation Link - https://doc.clickup.com/8631005/d/h/87cpx-293776/cd0f4122abc04d3
     */
    async createRelation(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/associations/relations', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get all relations By record Id
     * Get all relations by record Id
     */
    async getRelationsByRecordId(params, options) {
        const paramDefs = [{ name: 'recordId', in: 'path' }, { name: 'locationId', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'associationIds', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/associations/relations/{recordId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Relation
     * Delete Relation
     */
    async deleteRelation(params, options) {
        const paramDefs = [{ name: 'relationId', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/associations/relations/{relationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get association key by key name
     * Using this api you can get standard / user defined association by key
     */
    async getAssociationKeyByKeyName(params, options) {
        const paramDefs = [{ name: 'key_name', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/associations/key/{key_name}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get association by object keys
     * Get association by object keys like contacts, custom objects and opportunities. Documentation Link - https://doc.clickup.com/8631005/d/h/87cpx-293776/cd0f4122abc04d3
     */
    async getAssociationByObjectKeys(params, options) {
        const paramDefs = [{ name: 'objectKey', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/associations/objectKey/{objectKey}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Association By Id
     * Update Association , Allows you to update labels of an associations. Documentation Link - https://doc.clickup.com/8631005/d/h/87cpx-293776/cd0f4122abc04d3
     */
    async updateAssociation(params, requestBody, options) {
        const paramDefs = [{ name: 'associationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/associations/{associationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Association
     * Delete USER_DEFINED Association By Id, deleting an association will also all the relations for that association
     */
    async deleteAssociation(params, options) {
        const paramDefs = [{ name: 'associationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/associations/{associationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get association by ID
     * Using this api you can get SYSTEM_DEFINED / USER_DEFINED association by id
     */
    async getAssociationByID(params, options) {
        const paramDefs = [{ name: 'associationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/associations/{associationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Association
     * Allow you to create contact - contact , contact - custom objects associations, will add more in the future.Documentation Link - https://doc.clickup.com/8631005/d/h/87cpx-293776/cd0f4122abc04d3
     */
    async createAssociation(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/associations/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get all associations for a sub-account / location
     * Get all Associations
     */
    async findAssociations(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/associations/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Associations = Associations;
exports.default = Associations;
