// @gohighlevel/api-client/dist/lib/code/companies/companies.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Companies = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Companies Service
 * Documentation for Companies API
 */
class Companies {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Company
     * Get Comapny
     */
    async getCompany(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/companies/{companyId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Companies = Companies;
exports.default = Companies;
