// @gohighlevel/api-client/dist/lib/code/marketplace/marketplace.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Marketplace = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Marketplace Service
 * Documentation for Marketplace API
 */
class Marketplace {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Create a new wallet charge
     * Create a new wallet charge
     */
    async charge(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access-Only"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/marketplace/billing/charges', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get all wallet charges
     * Get all wallet charges
     */
    async getCharges(params, options) {
        const paramDefs = [{ name: 'meterId', in: 'query' }, { name: 'eventId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'startDate', in: 'query' }, { name: 'endDate', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access-Only"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/marketplace/billing/charges', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete a wallet charge
     * Delete a wallet charge
     */
    async deleteCharge(params, options) {
        const paramDefs = [{ name: 'chargeId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access-Only"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/marketplace/billing/charges/{chargeId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get specific wallet charge details
     * Get specific wallet charge details
     */
    async getSpecificCharge(params, options) {
        const paramDefs = [{ name: 'chargeId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access-Only"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/marketplace/billing/charges/{chargeId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Check if account has sufficient funds
     * Check if account has sufficient funds
     */
    async hasFunds(options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access-Only"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/marketplace/billing/charges/has-funds', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Uninstall an application
     * Uninstalls an application from your company or a specific location. This will remove the application&#x60;s access and stop all its functionalities
     */
    async uninstallApplication(params, requestBody, options) {
        const paramDefs = [{ name: 'appId', in: 'path' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access-Only", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/marketplace/app/{appId}/installations', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Installer Details
     * Fetches installer details for the authenticated user. This endpoint returns information about the company, location, user, and installation details associated with the current OAuth token.
     */
    async getInstallerDetails(params, options) {
        const paramDefs = [{ name: 'appId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = [];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/marketplace/app/{appId}/installations', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Marketplace = Marketplace;
exports.default = Marketplace;
