// @gohighlevel/api-client/dist/lib/code/forms/forms.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Forms = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Forms Service
 * Documentation for forms API
 */
class Forms {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Forms Submissions
     * Get Forms Submissions
     */
    async getFormsSubmissions(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'formId', in: 'query' }, { name: 'q', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/forms/submissions', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Upload files to custom fields
     * Post the necessary fields for the API to upload files. The files need to be a buffer with the key &quot;&lt; custom_field_id &gt;_&lt; file_id &gt;&quot;. &lt;br /&gt; Here custom field id is the ID of your custom field and file id is a randomly generated id (or uuid) &lt;br /&gt; There is support for multiple file uploads as well. Have multiple fields in the format mentioned.&lt;br /&gt;File size is limited to 50 MB.&lt;br /&gt;&lt;br /&gt; The allowed file types are: &lt;br/&gt; &lt;ul&gt;&lt;li&gt;PDF&lt;/li&gt;&lt;li&gt;DOCX&lt;/li&gt;&lt;li&gt;DOC&lt;/li&gt;&lt;li&gt;JPG&lt;/li&gt;&lt;li&gt;JPEG&lt;/li&gt;&lt;li&gt;PNG&lt;/li&gt;&lt;li&gt;GIF&lt;/li&gt;&lt;li&gt;CSV&lt;/li&gt;&lt;li&gt;XLSX&lt;/li&gt;&lt;li&gt;XLS&lt;/li&gt;&lt;li&gt;MP4&lt;/li&gt;&lt;li&gt;MPEG&lt;/li&gt;&lt;li&gt;ZIP&lt;/li&gt;&lt;li&gt;RAR&lt;/li&gt;&lt;li&gt;TXT&lt;/li&gt;&lt;li&gt;SVG&lt;/li&gt;&lt;/ul&gt; &lt;br /&gt;&lt;br /&gt; The API will return the updated contact object.
     */
    async uploadToCustomFields(params, requestBody, options) {
        const paramDefs = [{ name: 'contactId', in: 'query' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer", "Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/forms/upload-custom-files', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Forms
     * Get Forms
     */
    async getForms(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'type', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/forms/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Forms = Forms;
exports.default = Forms;
