// @gohighlevel/api-client/dist/lib/code/social-media-posting/social-media-posting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocialMediaPosting = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * SocialMediaPosting Service
 * Documentation for Social Media Posting API
 */
class SocialMediaPosting {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Starts OAuth For Google Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to Google login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;google&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch Google account details using below API -
    API: &#x27;/social-media-posting/oauth/google/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startGoogleOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/google/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get google business locations
     * Get google business locations
     */
    async getGoogleLocations(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/google/locations/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Set google business locations
     * Set google business locations
     */
    async setGoogleLocations(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/google/locations/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get posts
     * Get Posts
     */
    async getPosts(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts/list', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create post
     * Create posts for all supported platforms. It is possible to create customized posts per channel by using the same platform account IDs in a request and hitting the create post API multiple times with different summaries and account IDs per platform.
  
  The content and media limitations, as well as platform rate limiters corresponding to the respective platforms, are provided in the following reference link:
  
    Link: [Platform Limitations](https://help.leadconnectorhq.com/support/solutions/articles/48001240003-social-planner-image-video-content-and-api-limitations &quot;Social Planner Help&quot;)
     */
    async createPost(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get post
     * Get post
     */
    async getPost(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Edit post
     * Create posts for all supported platforms. It is possible to create customized posts per channel by using the same platform account IDs in a request and hitting the create post API multiple times with different summaries and account IDs per platform.
  
  The content and media limitations, as well as platform rate limiters corresponding to the respective platforms, are provided in the following reference link:
  
    Link: [Platform Limitations](https://help.leadconnectorhq.com/support/solutions/articles/48001240003-social-planner-image-video-content-and-api-limitations &quot;Social Planner Help&quot;)
     */
    async editPost(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Post
     * Delete Post
     */
    async deletePost(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Bulk Delete Social Planner Posts
     * Deletes multiple posts based on the provided list of post IDs.
                    This operation is useful for clearing up large numbers of posts efficiently.
                    
  Note:
                    
  1.The maximum number of posts that can be deleted in a single request is &#x27;50&#x27;.
                    
  2.However, It will only get deleted in Highlevel database but still
                     it is recommended to be cautious of this operation.
     */
    async bulkDeleteSocialPlannerPosts(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = [];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/posts/bulk-delete', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Accounts
     * Get list of accounts and groups
     */
    async getAccount(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/accounts', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Account
     * Delete account and account from group
     */
    async deleteAccount(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }, { name: 'companyId', in: 'query' }, { name: 'userId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/accounts/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For Facebook Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to Facebook login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;facebook&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch Facebook account details using below API -
    API: &#x27;/social-media-posting/oauth/facebook/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startFacebookOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/facebook/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get facebook pages
     * Get facebook pages
     */
    async getFacebookPageGroup(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/facebook/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Attach facebook pages
     * Attach facebook pages
     */
    async attachFacebookPageGroup(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/facebook/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For Instagram Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to Instagram login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;instagram&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch Instagram account details using below API -
    API: &#x27;/social-media-posting/oauth/instagram/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startInstagramOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/instagram/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Instagram Professional Accounts
     * Get Instagram Professional Accounts
     */
    async getInstagramPageGroup(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/instagram/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Attach Instagram Professional Accounts
     * Attach Instagram Professional Accounts
     */
    async attachInstagramPageGroup(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/instagram/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For LinkedIn Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to LinkedIn login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;linkedin&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch LinkedIn account details using below API -
    API: &#x27;/social-media-posting/oauth/linkedin/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startLinkedinOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/linkedin/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Linkedin pages and profile
     * Get Linkedin pages and profile
     */
    async getLinkedinPageProfile(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/linkedin/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Attach linkedin pages and profile
     * Attach linkedin pages and profile
     */
    async attachLinkedinPageProfile(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/linkedin/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For Twitter Account
     * &lt;div&gt;&lt;div&gt;
    &lt;span style&#x3D; &quot;display: inline-block;
      width: 25px; height: 25px;
      background-color: red;
      color: black;
      font-weight: bold;
      font-size: 24px;
      text-align: center;
      line-height: 20px;
      border: 2px solid black;
      border-radius: 20%;
      margin-right: 10px;&quot;&gt;
      !
    &lt;/span&gt;
    &lt;span&gt;&lt;strong&gt;As of December 4, 2024, X (formerly Twitter) is no longer supported. We apologise for any inconvenience.&lt;/strong&gt;&lt;/span&gt;
  &lt;/div&gt;&lt;/div&gt;
     * @deprecated This method is deprecated
     */
    async startTwitterOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/twitter/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Twitter profile
     * &lt;div&gt;&lt;div&gt;
    &lt;span style&#x3D; &quot;display: inline-block;
      width: 25px; height: 25px;
      background-color: red;
      color: black;
      font-weight: bold;
      font-size: 24px;
      text-align: center;
      line-height: 20px;
      border: 2px solid black;
      border-radius: 20%;
      margin-right: 10px;&quot;&gt;
      !
    &lt;/span&gt;
    &lt;span&gt;&lt;strong&gt;As of December 4, 2024, X (formerly Twitter) is no longer supported. We apologise for any inconvenience.&lt;/strong&gt;&lt;/span&gt;
  &lt;/div&gt;&lt;/div&gt;
     * @deprecated This method is deprecated
     */
    async getTwitterProfile(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = [];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/twitter/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Attach Twitter profile
     * &lt;div&gt;&lt;div&gt;
    &lt;span style&#x3D; &quot;display: inline-block;
      width: 25px; height: 25px;
      background-color: red;
      color: black;
      font-weight: bold;
      font-size: 24px;
      text-align: center;
      line-height: 20px;
      border: 2px solid black;
      border-radius: 20%;
      margin-right: 10px;&quot;&gt;
      !
    &lt;/span&gt;
    &lt;span&gt;&lt;strong&gt;As of December 4, 2024, X (formerly Twitter) is no longer supported. We apologise for any inconvenience.&lt;/strong&gt;&lt;/span&gt;
  &lt;/div&gt;&lt;/div&gt;
     * @deprecated This method is deprecated
     */
    async attachTwitterProfile(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = [];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/twitter/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Upload CSV
     *
     */
    async uploadCsv(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Upload Status
     *
     */
    async getUploadStatus(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'includeUsers', in: 'query' }, { name: 'userId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Set Accounts
     *
     */
    async setAccounts(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/set-accounts', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get CSV Post
     *
     */
    async getCsvPost(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Start CSV Finalize
     *
     */
    async startCsvFinalize(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete CSV
     *
     */
    async deleteCsv(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete CSV Post
     *
     */
    async deleteCsvPost(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'postId', in: 'path' }, { name: 'csvId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/csv/{csvId}/post/{postId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For Tiktok Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to Tiktok login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;tiktok&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch Tiktok account details using below API -
    API: &#x27;/social-media-posting/oauth/tiktok/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startTiktokOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/tiktok/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Tiktok profile
     * Get Tiktok profile
     */
    async getTiktokProfile(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/tiktok/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Attach Tiktok profile
     * Attach Tiktok profile
     */
    async attachTiktokProfile(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/tiktok/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Starts OAuth For Tiktok Business Account
     * Open the API in a window with appropriate params and headers instead of using the Curl. User is navigated to Tiktok-Business login OAuth screen. On successful login, listen on window object for message where event listener returns data in its callback function.
    ### Sample code to listen to event data:
      window.addEventListener(&#x27;message&#x27;,
        function(e) {
          if (e.data &amp;&amp; e.data.page &#x3D;&#x3D;&#x3D; &#x27;social_media_posting&#x27;) {
          const { actionType, page, platform, placement, accountId, reconnectAccounts } &#x3D; e.data
          }
        },
      false)
    ### Event Data Response:
      {
        actionType: string,            Ex: &quot;close&quot;
        page: string,                  Ex: &quot;social-media-posting&quot;
        platform: string,              Ex: &quot;tiktok-business&quot;
        placement: string,             Ex: &quot;placement&quot;
        accountId: string,             Ex: &quot;658a9b6833b91e0ecb8f3958&quot;
        reconnectAccounts: string[]]   Ex: [&quot;658a9b6833b91e0ecb834acd&quot;, &quot;efd2daa9b6833b91e0ecb8f3511&quot;]
      }
    ### The accountId retrieved from above data can be used to fetch Tiktok-Business account details using below API -
    API: &#x27;/social-media-posting/oauth/tiktok-business/accounts/:accountId&#x27;
  
    Method: GET
     */
    async startTiktokBusinessOauth(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'userId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'reconnect', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/tiktok-business/start', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Tiktok Business profile
     * Get Tiktok Business profile
     */
    async getTiktokBusinessProfile(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'accountId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/oauth/{locationId}/tiktok-business/accounts/{accountId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get categories by location id
     *
     */
    async getCategoriesLocationId(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'searchText', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'skip', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/categories', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get categories by id
     *
     */
    async getCategoriesId(params, options) {
        const paramDefs = [{ name: 'id', in: 'path' }, { name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/categories/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get tags by location id
     *
     */
    async getTagsLocationId(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'searchText', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'skip', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/tags', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get tags by ids
     *
     */
    async getTagsByIds(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/{locationId}/tags/details', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Social Media Statistics
     * Retrieve analytics data for multiple social media accounts. Provides metrics for the last 7 days with comparison to the previous 7 days. Supports filtering by platforms and specific connected accounts.
     */
    async getSocialMediaStatistics(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = [];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/social-media-posting/statistics', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.SocialMediaPosting = SocialMediaPosting;
exports.default = SocialMediaPosting;
