// @gohighlevel/api-client/dist/lib/code/emails/emails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Emails = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Emails Service
 * Documentation for emails API
 */
class Emails {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Campaigns
     * Get Campaigns
     */
    async fetchCampaigns(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }, { name: 'status', in: 'query' }, { name: 'emailStatus', in: 'query' }, { name: 'name', in: 'query' }, { name: 'parentId', in: 'query' }, { name: 'limitedFields', in: 'query' }, { name: 'archived', in: 'query' }, { name: 'campaignsOnly', in: 'query' }, { name: 'showStats', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/emails/schedule', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create a new template
     * Create a new template
     */
    async createTemplate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/emails/builder', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Fetch email templates
     * Fetch email templates by location id
     */
    async fetchTemplate(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }, { name: 'search', in: 'query' }, { name: 'sortByDate', in: 'query' }, { name: 'archived', in: 'query' }, { name: 'builderVersion', in: 'query' }, { name: 'name', in: 'query' }, { name: 'parentId', in: 'query' }, { name: 'originId', in: 'query' }, { name: 'templatesOnly', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/emails/builder', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete a template
     * Delete a template
     */
    async deleteTemplate(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'templateId', in: 'path' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/emails/builder/{locationId}/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update a template
     * Update a template
     */
    async updateTemplate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/emails/builder/data', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Emails = Emails;
exports.default = Emails;
