// @gohighlevel/api-client/dist/lib/code/phone-system/phone-system.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneSystem = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * PhoneSystem Service
 * Documentation for Phone System API
 */
class PhoneSystem {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * List Number Pools
     * Get list of number pools
     */
    async getNumberPoolList(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/phone-system/number-pools', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List active numbers
     * Retrieve a paginated list of active phone numbers for a specific location. Supports filtering, pagination, and optional exclusion of number pool assignments.
     */
    async activeNumbers(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'path' }, { name: 'pageSize', in: 'query' }, { name: 'page', in: 'query' }, { name: 'searchFilter', in: 'query' }, { name: 'skipNumberPool', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/phone-system/numbers/location/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.PhoneSystem = PhoneSystem;
exports.default = PhoneSystem;
