// @gohighlevel/api-client/dist/lib/code/surveys/surveys.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Surveys = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Surveys Service
 * Documentation for surveys API
 */
class Surveys {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Surveys Submissions
     * Get Surveys Submissions
     */
    async getSurveysSubmissions(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'page', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'surveyId', in: 'query' }, { name: 'q', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/surveys/submissions', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Surveys
     * Get Surveys
     */
    async getSurveys(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'type', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/surveys/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Surveys = Surveys;
exports.default = Surveys;
