// @gohighlevel/api-client/dist/lib/code/users/users.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Users Service
 * Documentation for users API
 */
class Users {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Search Users
     * Search Users
     */
    async searchUsers(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'query' }, { name: 'query', in: 'query' }, { name: 'skip', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'locationId', in: 'query' }, { name: 'type', in: 'query' }, { name: 'role', in: 'query' }, { name: 'ids', in: 'query' }, { name: 'sort', in: 'query' }, { name: 'sortDirection', in: 'query' }, { name: 'enabled2waySync', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/users/search', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Filter Users by Email
     * Filter users by company ID, deleted status, and email array
     */
    async filterUsersByEmail(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/users/search/filter-by-email', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get User
     * Get User
     */
    async getUser(params, options) {
        const paramDefs = [{ name: 'userId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/users/{userId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update User
     * Update User
     */
    async updateUser(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/users/{userId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete User
     * Delete User
     */
    async deleteUser(options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/users/{userId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get User by Location
     * Get User by Location
     */
    async getUserByLocation(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/users/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create User
     * Create User
     */
    async createUser(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Agency-Access", "Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/users/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Users = Users;
exports.default = Users;
