// @gohighlevel/api-client/dist/lib/code/funnels/funnels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Funnels = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Funnels Service
 * Documentation for funnels API
 */
class Funnels {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Create Redirect
     * The &quot;Create Redirect&quot; API Allows adding a new url redirect to the system. Use this endpoint to create a url redirect with the specified details. Ensure that the required information is provided in the request payload.
     */
    async createRedirect(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/funnels/lookup/redirect', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Redirect By Id
     * The &quot;Update Redirect By Id&quot; API Allows updating an existing URL redirect in the system. Use this endpoint to modify a URL redirect with the specified ID using details provided in the request payload.
     */
    async updateRedirectById(params, requestBody, options) {
        const paramDefs = [{ name: 'id', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/funnels/lookup/redirect/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Redirect By Id
     * The &quot;Delete Redirect By Id&quot; API Allows deletion of a URL redirect from the system using its unique identifier. Use this endpoint to delete a URL redirect with the specified ID using details provided in the request payload.
     */
    async deleteRedirectById(params, options) {
        const paramDefs = [{ name: 'id', in: 'path' }, { name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/funnels/lookup/redirect/{id}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Fetch List of Redirects
     * Retrieves a list of all URL redirects based on the given query parameters.
     */
    async fetchRedirectsList(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }, { name: 'search', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/funnels/lookup/redirect/list', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Fetch List of Funnels
     * Retrieves a list of all funnels based on the given query parameters.
     */
    async getFunnels(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'type', in: 'query' }, { name: 'category', in: 'query' }, { name: 'offset', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'parentId', in: 'query' }, { name: 'name', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/funnels/funnel/list', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Fetch list of funnel pages
     * Retrieves a list of all funnel pages based on the given query parameters.
     */
    async getPagesByFunnelId(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'funnelId', in: 'query' }, { name: 'name', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/funnels/page', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Fetch count of funnel pages
     * Retrieves count of all funnel pages based on the given query parameters.
     */
    async getPagesCountByFunnelId(params, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }, { name: 'funnelId', in: 'query' }, { name: 'name', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/funnels/page/count', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Funnels = Funnels;
exports.default = Funnels;
