// @gohighlevel/api-client/dist/lib/code/email-isv/email-isv.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailIsv = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * EmailIsv Service
 * Documentation for Email ISV API
 */
class EmailIsv {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Email Verification
     * Verify Email
     */
    async verifyEmail(params, requestBody, options) {
        const paramDefs = [{ name: 'locationId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/email/verify', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.EmailIsv = EmailIsv;
exports.default = EmailIsv;
