// @gohighlevel/api-client/dist/lib/code/snapshots/snapshots.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Snapshots = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Snapshots Service
 * Documentation for Snapshots API
 */
class Snapshots {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Get Snapshots
     * Get a list of all own and imported Snapshots
     */
    async getCustomSnapshots(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/snapshots/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Snapshot Share Link
     * Create a share link for snapshot
     */
    async createSnapshotShareLink(params, requestBody, options) {
        const paramDefs = [{ name: 'companyId', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/snapshots/share/link', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Snapshot Push between Dates
     * Get list of sub-accounts snapshot pushed in time period
     */
    async getSnapshotPush(params, options) {
        const paramDefs = [{ name: 'snapshotId', in: 'path' }, { name: 'companyId', in: 'query' }, { name: 'from', in: 'query' }, { name: 'to', in: 'query' }, { name: 'lastDoc', in: 'query' }, { name: 'limit', in: 'query' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/snapshots/snapshot-status/{snapshotId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get Last Snapshot Push
     * Get Latest Snapshot Push Status for a location id
     */
    async getLatestSnapshotPush(params, options) {
        const paramDefs = [{ name: 'companyId', in: 'query' }, { name: 'snapshotId', in: 'path' }, { name: 'locationId', in: 'path' },];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/snapshots/snapshot-status/{snapshotId}/location/{locationId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {});
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Snapshots = Snapshots;
exports.default = Snapshots;
