// @gohighlevel/api-client/dist/lib/logging/log-level.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogLevel = void 0;
exports.parseLogLevel = parseLogLevel;
/**
 * Log levels for the GHL SDK
 * Higher numbers include all lower level logs
 */
var LogLevel;
(function (LogLevel) {
    LogLevel[LogLevel["NONE"] = 0] = "NONE";
    LogLevel[LogLevel["ERROR"] = 1] = "ERROR";
    LogLevel[LogLevel["WARN"] = 2] = "WARN";
    LogLevel[LogLevel["INFO"] = 3] = "INFO";
    LogLevel[LogLevel["DEBUG"] = 4] = "DEBUG"; // All logs (most verbose)
})(LogLevel || (exports.LogLevel = LogLevel = {}));
/**
 * Parse string log level to enum value
 * @param level - String log level
 * @returns LogLevel enum value
 */
function parseLogLevel(level) {
    switch (level.toLowerCase()) {
        case 'none': return LogLevel.NONE;
        case 'error': return LogLevel.ERROR;
        case 'warn': return LogLevel.WARN;
        case 'info': return LogLevel.INFO;
        case 'debug': return LogLevel.DEBUG;
        default: return LogLevel.WARN; // Default fallback
    }
}
