// @gohighlevel/api-client/dist/lib/code/invoices/invoices.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Invoices = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Invoices Service
 * Documentation for invoice API
 */
class Invoices {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Create template
     * API to create a template
     */
    async createInvoiceTemplate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/template', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List templates
     * API to get list of templates
     */
    async listInvoiceTemplates(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'status', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }, { name: 'search', in: 'query' }, { name: 'paymentMode', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/template', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get an template
     * API to get an template by template id
     */
    async getInvoiceTemplate(params, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/template/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update template
     * API to update an template by template id
     */
    async updateInvoiceTemplate(params, requestBody, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/invoices/template/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete template
     * API to update an template by template id
     */
    async deleteInvoiceTemplate(params, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/invoices/template/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update template late fees configuration
     * API to update template late fees configuration by template id
     */
    async updateInvoiceTemplateLateFeesConfiguration(params, requestBody, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/invoices/template/{templateId}/late-fees-configuration', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update template late fees configuration
     * API to update template late fees configuration by template id
     */
    async updateInvoicePaymentMethodsConfiguration(params, requestBody, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/invoices/template/{templateId}/payment-methods-configuration', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Invoice Schedule
     * API to create an invoice Schedule
     */
    async createInvoiceSchedule(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List schedules
     * API to get list of schedules
     */
    async listInvoiceSchedules(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'status', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }, { name: 'search', in: 'query' }, { name: 'paymentMode', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get an schedule
     * API to get an schedule by schedule id
     */
    async getInvoiceSchedule(params, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update schedule
     * API to update an schedule by schedule id
     */
    async updateInvoiceSchedule(params, requestBody, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete schedule
     * API to delete an schedule by schedule id
     */
    async deleteInvoiceSchedule(params, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update scheduled recurring invoice
     * API to update scheduled recurring invoice
     */
    async updateAndScheduleInvoiceSchedule(params, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}/updateAndSchedule', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Schedule an schedule invoice
     * API to schedule an schedule invoice to start sending to the customer
     */
    async scheduleInvoiceSchedule(params, requestBody, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}/schedule', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Manage Auto payment for an schedule invoice
     * API to manage auto payment for a schedule
     */
    async autoPaymentInvoiceSchedule(params, requestBody, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}/auto-payment', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Cancel an scheduled invoice
     * API to cancel a scheduled invoice by schedule id
     */
    async cancelInvoiceSchedule(params, requestBody, options) {
        const paramDefs = [{ name: 'scheduleId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/schedule/{scheduleId}/cancel', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create &amp; Send
     * API to create or update a text2pay invoice
     */
    async text2payInvoice(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/text2pay', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Generate Invoice Number
     * Get the next invoice number for the given location
     */
    async generateInvoiceNumber(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/generate-invoice-number', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Get invoice
     * API to get invoice by invoice id
     */
    async getInvoice(params, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update invoice
     * API to update invoice by invoice id
     */
    async updateInvoice(params, requestBody, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete invoice
     * API to delete invoice by invoice id
     */
    async deleteInvoice(params, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }, { name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update invoice late fees configuration
     * API to update invoice late fees configuration by invoice id
     */
    async updateInvoiceLateFeesConfiguration(params, requestBody, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}/late-fees-configuration', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Void invoice
     * API to delete invoice by invoice id
     */
    async voidInvoice(params, requestBody, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}/void', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Send invoice
     * API to send invoice by invoice id
     */
    async sendInvoice(params, requestBody, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}/send', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Record a manual payment for an invoice
     * API to record manual payment for an invoice by invoice id
     */
    async recordInvoice(params, requestBody, options) {
        const paramDefs = [{ name: 'invoiceId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/{invoiceId}/record-payment', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update invoice last visited at
     * API to update invoice last visited at by invoice id
     */
    async updateInvoiceLastVisitedAt(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/invoices/stats/last-visited-at', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create New Estimate
     * Create a new estimate with the provided details
     */
    async createNewEstimate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Estimate
     * Update an existing estimate with new details
     */
    async updateEstimate(params, requestBody, options) {
        const paramDefs = [{ name: 'estimateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/{estimateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Estimate
     * Delete an existing estimate
     */
    async deleteEstimate(params, requestBody, options) {
        const paramDefs = [{ name: 'estimateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/{estimateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Generate Estimate Number
     * Get the next estimate number for the given location
     */
    async generateEstimateNumber(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/number/generate', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Send Estimate
     * API to send estimate by estimate id
     */
    async sendEstimate(params, requestBody, options) {
        const paramDefs = [{ name: 'estimateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/{estimateId}/send', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Invoice from Estimate
     * Create a new invoice from an existing estimate
     */
    async createInvoiceFromEstimate(params, requestBody, options) {
        const paramDefs = [{ name: 'estimateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/{estimateId}/invoice', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List Estimates
     * Get a paginated list of estimates
     */
    async listEstimates(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }, { name: 'search', in: 'query' }, { name: 'status', in: 'query' }, { name: 'contactId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/list', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update estimate last visited at
     * API to update estimate last visited at by estimate id
     */
    async updateEstimateLastVisitedAt(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PATCH',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/stats/last-visited-at', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List Estimate Templates
     * Get a list of estimate templates or a specific template by ID
     */
    async listEstimateTemplates(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'search', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/template', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Estimate Template
     * Create a new estimate template
     */
    async createEstimateTemplate(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/template', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Update Estimate Template
     * Update an existing estimate template
     */
    async updateEstimateTemplate(params, requestBody, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'PUT',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/template/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Delete Estimate Template
     * Delete an existing estimate template
     */
    async deleteEstimateTemplate(params, requestBody, options) {
        const paramDefs = [{ name: 'templateId', in: 'path' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'DELETE',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/template/{templateId}', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Preview Estimate Template
     * Get a preview of an estimate template
     */
    async previewEstimateTemplate(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'templateId', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/estimate/template/preview', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * Create Invoice
     * API to create an invoice
     */
    async createInvoice(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/invoices/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
    /**
     * List invoices
     * API to get list of invoices
     */
    async listInvoices(params, options) {
        const paramDefs = [{ name: 'altId', in: 'query' }, { name: 'altType', in: 'query' }, { name: 'status', in: 'query' }, { name: 'startAt', in: 'query' }, { name: 'endAt', in: 'query' }, { name: 'search', in: 'query' }, { name: 'paymentMode', in: 'query' }, { name: 'contactId', in: 'query' }, { name: 'limit', in: 'query' }, { name: 'offset', in: 'query' }, { name: 'sortField', in: 'query' }, { name: 'sortOrder', in: 'query' }];
        const extracted = (0, request_utils_1.extractParams)(params, paramDefs);
        const requirements = ["Location-Access", "Agency-Access"];
        const config = {
            method: 'GET',
            url: (0, request_utils_1.buildUrl)('/invoices/', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            __secutiryRequirements: requirements,
            __preferredTokenType: options?.preferredTokenType,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, {}, options?.preferredTokenType);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Invoices = Invoices;
exports.default = Invoices;
