// @gohighlevel/api-client/dist/lib/code/courses/courses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Courses = void 0;
const request_utils_1 = require("../../utils/request-utils");
/**
 * Courses Service
 * API Service for Courses and Memberships
 */
class Courses {
    constructor(httpClient) {
        this.client = httpClient;
    }
    /**
     * Import Courses
     * Import Courses through public channels
     */
    async importCourses(requestBody, options) {
        const paramDefs = [];
        const extracted = (0, request_utils_1.extractParams)(null, paramDefs);
        const requirements = ["bearer"];
        const config = {
            method: 'POST',
            url: (0, request_utils_1.buildUrl)('/courses/courses-exporter/public/import', extracted.path),
            params: extracted.query,
            headers: { ...extracted.header, ...options?.headers },
            data: requestBody,
            __secutiryRequirements: requirements,
            __pathParams: extracted.path,
            ...options
        };
        const authToken = await (0, request_utils_1.getAuthToken)(this.client, requirements, config.headers || {}, { ...config.params || {}, ...config.__pathParams }, requestBody);
        if (authToken) {
            config.headers = { ...config.headers, Authorization: authToken };
        }
        const response = await this.client.request(config);
        return response.data;
    }
}
exports.Courses = Courses;
exports.default = Courses;
